### PR TITLE
C++: Model `scanf` and `fscanf` as flow sources

### PR DESCRIPTION
**Pull request by jketema**

### CODE REVIEW

File: cpp/ql/lib/semmle/code/cpp/models/implementations/Scanf.qll [hunk 1]
```diff
@@ -70,3 +67,40 @@ private class SscanfModel extends ArrayFunction, TaintFunction, AliasFunction, S
       ]
   }
 }
+
+/**
+ * The standard function `scanf` and its assorted variants
+ */
+private class ScanfModel extends ScanfFunctionModel, LocalFlowSourceFunction {
+  ScanfModel() { this instanceof Scanf }
```
Comment (MathiasVP) [inline cpp/ql/lib/semmle/code/cpp/models/implementations/Scanf.qll:75]: Might as well be nice and help out to the compiler 😂
```suggestion
private class ScanfModel extends ScanfFunctionModel, LocalFlowSourceFunction instanceof Scanf {
```

File: cpp/ql/lib/semmle/code/cpp/models/implementations/Scanf.qll [hunk 2]
```diff
@@ -70,3 +67,40 @@ private class SscanfModel extends ArrayFunction, TaintFunction, AliasFunction, S
...
+    output.isParameterDeref(any(int i | i >= this.getArgsStartPosition())) and
+    description = "Value read by " + this.getName()
+  }
+}
+
+/**
+ * The standard function `fscanf` and its assorted variants
+ */
+private class FscanfModel extends ScanfFunctionModel, RemoteFlowSourceFunction {
+  FscanfModel() { this instanceof Fscanf }
```
Comment (MathiasVP) [inline cpp/ql/lib/semmle/code/cpp/models/implementations/Scanf.qll:87]:
```suggestion
private class FscanfModel extends ScanfFunctionModel, RemoteFlowSourceFunction instanceof Fscanf {
```

File: cpp/ql/lib/semmle/code/cpp/models/implementations/Scanf.qll (hunk 1 above)
Comment (jketema) [reply cpp/ql/lib/semmle/code/cpp/models/implementations/Scanf.qll]: Fixed

Comment (MathiasVP) [reply cpp/ql/lib/semmle/code/cpp/models/implementations/Scanf.qll]: > We really need a QL-for-QL check for this.

Agreed 👍

**Review by MathiasVP — APPROVED** [review]
LGTM!

**Pull request merged**